build.gradle

```
@@ -60,15 +63,47 @@ repositories {
 }
 
 
-configurations.all {
+configurations.configureEach {
     exclude group: "edu.wpi.first.wpilibj"
 }
 
-task(checkAkitInstall, dependsOn: "classes", type: JavaExec) {
+tasks.register('checkAkitInstall', JavaExec) {
+    dependsOn "classes"
     mainClass = "org.littletonrobotics.junction.CheckInstall"
     classpath = sourceSets.main.runtimeClasspath
 }
-compileJava.finalizedBy checkAkitInstall
+
+tasks.register('autoCommit') {
+    doLast {
+        def taskNames = project.gradle.startParameter.taskNames
+        if (taskNames.any({ it.toLowerCase().contains("deploy") }) || taskNames.any({ it.toLowerCase().contains("simulate") })) {
+
+            def eventPrefix = "event"
+            def fieldPrefix = "field"
+            def checkPrefix = "check"
+            def branch = 'git branch --show-current'.execute().text.trim()
+            def commitMessage = "autoCommit - updated at '${new Date().toString()}'"
+
+            if (branch.startsWith(eventPrefix) || branch.startsWith(fieldPrefix) || branch.startsWith(checkPrefix)) {
+                exec {
+                    workingDir(projectDir)
+                    executable('git')
+                    args 'commit', '-am', commitMessage
+                    ignoreExitValue = true
+                }
+
+                println("commited to branch -> '$branch'")
+                println("commited message -> '$commitMessage'")
+            } else {
+                println("skipping -> not a branch to auto commit")
+            }
+        } else {
+            println("skipping -> not a deploy or simulate")
+        }
```

instead of doing else at the end, use the inverse conditions put it first to make it cleaner.
```suggestion
        if (!taskNames.any({ it.toLowerCase().contains("deploy") }) && !taskNames.any({ it.toLowerCase().contains("simulate") })) {
            println("skipping -> not a deploy or simulate")
            return
        }

        def eventPrefix = "event"
        def fieldPrefix = "field"
        def checkPrefix = "check"
        def branch = 'git branch --show-current'.execute().text.trim()
        def commitMessage = "autoCommit - updated at '${new Date().toString()}'"

        if (!branch.startsWith(eventPrefix) && !branch.startsWith(fieldPrefix) && !branch.startsWith(checkPrefix)) {
            println("skipping -> not a branch to auto commit")
            return
        }
        exec {
            workingDir(projectDir)
            executable('git')
            args 'commit', '-am', commitMessage
            ignoreExitValue = true
        }

        println("commited to branch -> '$branch'")
        println("commited message -> '$commitMessage'")
```

---

Why , it is making the if more complicated to understand